core/src/test/java/org/elasticsearch/index/query/RangeQueryBuilderTests.java

```diff
@@ -399,317 +397,13 @@ public void testNamedQueryParsing() throws IOException {
-        query.from(queryFromValue);
-        query.to(queryToValue);
-        QueryShardContext queryShardContext = queryShardContext();
-        FieldStatsProvider fieldStatsProvider = new FieldStatsProvider(null, null) {
-
-            @Override
-            public Relation isFieldWithinQuery(String fieldName, Object from, Object to, boolean includeLower, boolean includeUpper,
-                    DateTimeZone timeZone, DateMathParser dateMathParser) throws IOException {
-                return Relation.DISJOINT;
-            }
-        };
-        queryShardContext.setFieldStatsProvider(fieldStatsProvider);
-        QueryBuilder<?> rewritten = query.rewrite(queryShardContext);
-        assertThat(rewritten, instanceOf(MatchNoneQueryBuilder.class));
-    }
-
-    public void testRewriteFloatToSame() throws IOException {
-        String fieldName = randomAsciiOfLengthBetween(1, 20);
-        RangeQueryBuilder query = new RangeQueryBuilder(fieldName);
-        float queryFromValue = (float) randomDoubleBetween(-1000000, 1000000, true);
-        float queryToValue = (float) randomDoubleBetween(queryFromValue, 2000000, true);
-        query.from(queryFromValue);
-        query.to(queryToValue);
-        QueryShardContext queryShardContext = queryShardContext();
-        FieldStatsProvider fieldStatsProvider = new FieldStatsProvider(null, null) {
-
-            @Override
-            public Relation isFieldWithinQuery(String fieldName, Object from, Object to, boolean includeLower, boolean includeUpper,
-                    DateTimeZone timeZone, DateMathParser dateMathParser) throws IOException {
-                return Relation.INTERSECTS;
-            }
-        };
-        queryShardContext.setFieldStatsProvider(fieldStatsProvider);
-        QueryBuilder<?> rewritten = query.rewrite(queryShardContext);
-        assertThat(rewritten, sameInstance(query));
-    }
-
-    public void testRewriteTextToMatchAll() throws IOException {
+    public void testRewriteDateToMatchAll() throws IOException {
         String fieldName = randomAsciiOfLengthBetween(1, 20);
-        RangeQueryBuilder query = new RangeQueryBuilder(fieldName);
-        String queryFromValue = "damson";
-        String queryToValue = "plum";
-        String shardMinValue = "grape";
-        String shardMaxValue = "orange";
-        query.from(queryFromValue);
-        query.to(queryToValue);
-        QueryShardContext queryShardContext = queryShardContext();
-        FieldStatsProvider fieldStatsProvider = new FieldStatsProvider(null, null) {
-
-            @Override
-            public Relation isFieldWithinQuery(String fieldName, Object from, Object to, boolean includeLower, boolean includeUpper,
-                    DateTimeZone timeZone, DateMathParser dateMathParser) throws IOException {
+        RangeQueryBuilder query = new RangeQueryBuilder(fieldName) {
+            protected MappedFieldType.Relation getRelation(QueryRewriteContext queryRewriteContext) throws IOException {
```

Comment:
I'm not a fan of overriding the class under test, but short of mocking IndexReader and MapperService I'm not sure how we can test this here.